pkg/ice/agent.go

```diff
@@ -166,10 +169,12 @@ func (a *Agent) pingCandidate(local, remote Candidate) {
 }
 
 func (a *Agent) updateConnectionState(newState ConnectionState) {
-	a.connectionState = newState
```

Comment:
Can you remove the check here https://github.com/pions/webrtc/blob/master/rtcpeerconnection.go#L786 so we don't have logic to de-dupe in two places.

thanks!